### ISSUE TITLE
Bug: wrong ImportAttribute#parent type

### ISSUE THREAD

**Issue by ota-meshi** [typescript-eslint/typescript-eslint#10254]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&showAST=es&fileType=.tsx&code=JYWwDg9gTgLgBAQzgMyhEcDkA6A9AzOAd2BgAs4BvGATzAFMAuLAKwGcIA7TAXwG4AUPQAekWFTgAjODxRoMOXJMIlyVWg2aZ2XXoJFj4AKjnoseAMYrSFanSasO3fgNCGAFIoAmmADQTVMmZKOA0HbSdCWR4ASkEgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

Looking at the source code, the types are currently defined as follows:

```ts
  interface ImportAttribute {
    parent: TSESTree.ImportDeclaration | TSESTree.ImportExpression;
  }
```

https://github.com/typescript-eslint/typescript-eslint/blob/964fa91d5744ae3d6377618c74609550489ef766/packages/types/src/ts-estree.ts#L54

However, I think the correct way to follow ESTree is as follows:

```diff
  interface ImportAttribute {
-    parent: TSESTree.ImportDeclaration | TSESTree.ImportExpression;
+    parent: TSESTree.ImportDeclaration | TSESTree. ExportNamedDeclaration | TSESTree. ExportAllDeclaration;
  }
```

https://github.com/estree/estree/blob/master/es2025.md

What do you think?

### ESLint Config

```javascript
module.exports = {
  /* This is not necessary. */
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // This is not necessary.
  }
}
```

### Expected Result

I think `ImportAttribute#parent` needs to be `TSESTree.ImportDeclaration | TSESTree. ExportNamedDeclaration | TSESTree. ExportAllDeclaration`.

### Actual Result

Currently, `ImportAttribute#parent` is `TSESTree.ImportDeclaration | TSESTree.ImportExpression`, but I think this is wrong.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.12.2` |
| `@typescript-eslint/parser`            | `8.12.2` |
| `@typescript-eslint/rule-tester`       | `8.12.2` |
| `@typescript-eslint/scope-manager`     | `8.12.2` |
| `@typescript-eslint/typescript-estree` | `8.12.2` |
| `@typescript-eslint/type-utils`        | `8.12.2` |
| `@typescript-eslint/utils`             | `8.12.2` |
| `TypeScript`                           | `5.6.2` |
| `ESLint`                               | `9.3.0` |
| `node`                                 | `22.4.1` |
